tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/permission/TasklistPermissionServices.java

```diff
@@ -12,54 +12,72 @@
 
   private final SecurityConfiguration securityConfiguration;
-  private final AuthorizationChecker authorizationChecker;
   private final SecurityContextProvider securityContextProvider;
+  private final AuthorizationChecker authorizationChecker;
 
   public TasklistPermissionServices(
       final SecurityConfiguration securityConfiguration,
-      final AuthorizationChecker authorizationChecker,
-      final SecurityContextProvider securityContextProvider) {
+      final SecurityContextProvider securityContextProvider,
+      final AuthorizationChecker authorizationChecker) {
     this.securityConfiguration = securityConfiguration;
-    this.authorizationChecker = authorizationChecker;
     this.securityContextProvider = securityContextProvider;
+    this.authorizationChecker = authorizationChecker;
   }
 
   public boolean hasPermissionToCreateProcessInstance(final String bpmnProcessId) {
-    final var authentication = RequestMapper.getAuthentication();
-    return isAuthorizedForResource(bpmnProcessId, authentication, CREATE_PROC_INST_AUTH_CHECK);
+    return isAuthorizedForResource(bpmnProcessId, CREATE_PROC_INST_AUTH_CHECK);
+  }
+
+  public boolean hasPermissionToReadProcessDefinition(final String bpmnProcessId) {
+    return isAuthorizedForResource(bpmnProcessId, READ_PROC_DEF_AUTH_CHECK);
   }
 
   public boolean hasPermissionToUpdateUserTask(final TaskEntity task) {
+    return isAuthorizedForResource(task.getBpmnProcessId(), UPDATE_USER_TASK_AUTH_CHECK);
+  }
+
+  public List<String> getProcessDefinitionsWithCreateProcessInstancePermission() {
+    if (isAuthorizationDisabled()) {
+      return WILD_CARD_PERMISSION;
+    }
     final var authentication = RequestMapper.getAuthentication();
-    return isAuthorizedForResource(
-        task.getBpmnProcessId(), authentication, UPDATE_USER_TASK_AUTH_CHECK);
+    if (isWithoutAuthenticatedUserKey(authentication)) {
+      return WILD_CARD_PERMISSION;
+    }
+
+    final var securityContext =
+        securityContextProvider.provideSecurityContext(authentication, CREATE_PROC_INST_AUTH_CHECK);
+    return authorizationChecker.retrieveAuthorizedResourceKeys(securityContext);
   }
 
   private boolean isAuthorizedForResource(
-      final String resourceId,
-      final Authentication authentication,
-      final Authorization authorization) {
+      final String resourceId, final Authorization authorization) {
 
+    final var authentication = RequestMapper.getAuthentication();
     if (isAuthorizationCheckDisabled(authentication)) {
       return true;
     }
 
-    final var securityContext =
-        securityContextProvider.provideSecurityContext(authentication, authorization);
-    return authorizationChecker.isAuthorized(resourceId, securityContext);
+    return securityContextProvider.isAuthorized(resourceId, authentication, authorization);
   }
 
   private boolean isAuthorizationCheckDisabled(final Authentication authentication) {
```

Comment:
can we have lazy initialization of the Authentication (not needed in case authorizations are disabled)
```suggestion
  private boolean isAuthorizationCheckDisabled(final Supplier<Authentication> authentication) {
```